packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -772,6 +803,63 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
         return iconClass.concat(additionalClasses).join(' ');
     }
 
+    /**
+     * Add the parent id of the nodes that need tree indent guideline into a set
+     * @param node the tree node.
+     */
+    protected addActiveNodeParent(node: TreeNode): void {
+        if (this.isExpandable(node)) {
+            if (node.children && node.children.length > 0) {
+                this.parentOfActiveIndentGuideline.add(node.id);
+            }
+        } else {
+            const parent = node.parent;
+            if (parent && parent.children.length > 0) {
+                this.parentOfActiveIndentGuideline.add(parent.id);
+            }
+        }
+    }
+
+    /**
+     * Check to see if the indent's corresponding parent node is in the set
+     * @param node the tree node.
+     * @param currentLevel the level of the indent guide that is currently being rendered.
+     * @param depth the depth of the tree node.
+     */
+    protected isIndentsParentInSet(node: TreeNode, currentLevel: number, depth: number): boolean {
```

Comment:
Isn't it easier to put the tree model walk up inside the same loop `renderIndent` does?